rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -186,7 +192,12 @@ void SequentialWriter::close()
     // bag file was closed before callback call.
     callback_manager_.execute_callbacks(bag_events::BagEvent::WRITE_SPLIT, info);
   }
-  storage_factory_.reset();
+
+  topics_names_to_info_.clear();
+  topic_names_to_message_definitions_.clear();
+
+  converter_.reset();
+  message_definitions_.reset();
```

Comment:
```suggestion
```

The   `converter_` and `message_definitions_.` is unrelated for the fix and will be beneficial to let them persist over open/close operations.

Reply:
We actually need to reset the converter on close. I was wrong.
The message_definitions_  was addressed in the commit https://github.com/ros2/rosbag2/pull/1599/commits/5a1df9e50c6b10964b4b73951613b5f7b4a8c144